### ISSUE TITLE
Use SEMI JOINs for Binary Operators

### ISSUE THREAD

**Issue by aguynamedryan** [outcomesinsights/conceptql#170]
We're looking to swap out a lot of the JOINs we're doing with SEMI JOINs.

Most of the time, we only want to find a single copy of a LHS result, even if there are multiple RHS results that match.

As best I can tell, both Impala and PostgreSQL will use a SEMI JOIN if given this kind of syntax:

```
SELECT *
FROM left_table l
WHERE EXISTS(
  SELECT 1
  FROM right_table r
  WHERE l.person_id = r.person_id
  AND l.start_date BETWEEN r.start_date AND r.end_date)
```

There are still rare occasions where we'll want to do a proper JOIN that will create an LHS result for each RHS result that matches.  So we don't want to remove the old way of doing JOINs, we just want to default to SEMI JOIN and allow some parameter passed to a binary operator to indicate if we want to use an INNER JOIN instead.

**Comment by jeremyevans** [comment]
This conversion will need to be done on a per-operator basis. I'll try converting a few operators and see if there is a way to make the handling generic, so that the per-operator changes are minimal.

I'll only have about a half hour to work on this tonight, but hopefully have more time tomorrow night, and possible be able to spend time Friday morning.  I'm going on vacation Friday at noon and won't be able to work on this over the weekend.

**Comment by aguynamedryan** [comment]
It look like this has been implemented in 29faa10f14f0b19223647f7d5a5a9b2032a11309
